### PR TITLE
add option to disable group mappings

### PR DESCRIPTION
**Pull request by EmTeedee**
Hi,

adds the option to disable group mappings (and the ldap search request).
Defaults to have them enabled (for backward compatibility).

EmTeedee

### CODE REVIEW

File: authLdap.php [hunk 1]
```diff
@@ -210,44 +216,55 @@ function authLdap_login($user, $username, $password, $already_md5 = false)
...
-                break;
+            authldap_debug('LDAP groups: ' . json_encode($grp));
+
+            // Check whether the user is member of one of the groups that are
+            // allowed acces to the blog. If the user is not member of one of
+            // The groups throw her out! ;-)
+            // If the user is member of more than one group only the first one
+            // will be taken into account!
+
+            if ($authLDAPGroupEnable) {
```
Comment (heiglandreas) [inline authLdap.php]: Isn't that a duplicate from line 225?

File: view/admin.phtml [hunk 2]
```diff
@@ -200,6 +220,18 @@
...
             <table class="form-table">
                 <tr>
+                    <th>
+                        <label for="authLDAPGroupEnable">Use LDAP Groups to map to Roles?</label>
+                    </th>
+                    <td>
+                        <input type="hidden" name="authLDAPGroupEnable" value="0"/>
+                        <input type='checkbox' name='authLDAPGroupEnable' value='1'<?php echo $tGroupChecked; ?>/>
+                        <p class="description">
+                            Search LDAP for user's groups an map to Wordpress Roles.
```
Comment (heiglandreas) [inline view/admin.phtml]: ... an**d** map ...

**Comment by heiglandreas** [comment]
Great work! Thank you! Tehre are two minor issues I'd like to be fixed though :wink:

**Comment by chaplina** [comment]
Good stuff, but there are problems.

IMHO, it might be better to put the group stuff into a function and then 
only call that function if the group option is enabled.

**Comment by EmTeedee** [comment]
This should address all the problems raised.

**Comment by heiglandreas** [comment]
Thanks! Looking good :+1:

**Pull request merged**